NotesApi/V1/Gateways/DynamoDbGateway.cs

```diff
@@ -18,15 +20,32 @@ public DynamoDbGateway(IDynamoDBContext dynamoDbContext)
             _dynamoDbContext = dynamoDbContext;
         }
 
-        public List<Entity> GetAll()
+        // This method cannot be unit tested because of having to use GetTargetTable()
+        // which returns an unmockable concrete class.
+        // See here: https://github.com/aws/aws-sdk-net/issues/1310
+        [ExcludeFromCodeCoverage]
+        public async Task<List<Note>> GetByTargetIdAsync(Guid targetId)
         {
-            return new List<Entity>();
-        }
+            List<NoteDb> dbNotes = new List<NoteDb>();
 
-        public Entity GetEntityById(int id)
-        {
-            var result = _dynamoDbContext.LoadAsync<DatabaseEntity>(id).GetAwaiter().GetResult();
-            return result?.ToDomain();
+            // We query directly on the Table (rather than using _dynamoDbContext.QueryAsync())
```

Comment:
I would try to reduce as much as possible the comments in the code as it should document itself.  I can understand the comment above due a not obvious issue with the sdk but I would probably remove this one as that how to query docuements via partitionkey. What other people think about it?

Reply:
I agree with trying not to use comments as a general rule. 
The only reason there is one here is because, as you acknowledge, there is a very particular reason why were are having to use a non-obvious implementation, That said if people think it is not necessary I'm not precious about it.

Reply:
Removed